spec/message.abnf

```diff
@@ -60,7 +60,7 @@ private-start  = "^" / "&"
 ; future versions of this specification
 reserved       = reserved-start reserved-body
 reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
-reserved-body  = *( [s] 1*(reserved-char / reserved-escape / quoted))
+reserved-body  = name *(s 1*(reserved-char / reserved-escape / quoted))
```

Comment:
I hope it is non-controversial to reject expressions like `{@}` and `{^|foo|}` and `{& bar}` as syntactically invalid.

Reply:
This would preclude the possibility of using reserved placeholders as inline comments. For example:
```
The meeting repeats fortnightly {? every second week ?} on Thursdays
```
Or as protection markers:
```
This text is {<}<b>{>}bold{<}</b>{>}
```

I'm not suggesting that we should ever do either of the above, but that restricting `reserved-body` like this would need to be explicitly discussed.

Reply:
Absolutely! If we do want to preserve that possibility, I think I would suggest a slightly different refactor:
```diff
-reserved-body  = *( [s] 1*(reserved-char / reserved-escape / quoted))
+reserved-body       = [s] [(1*name-char / reserved-body-start) *(s reserved-body-part) [s]]
+reserved-body-start = *name-char (non-name-char / reserved-escape / quoted) [reserved-body-part]
+reserved-body-part  = 1*(reserved-char / reserved-escape / quoted)
+non-name-char       = %x21-2C        ; omit - .
+                    / %x2F           ; omit digits and :
+                    / %x3B-40        ; omit uppercase letters
+                    / %x5B           ; omit \
+                    / %x5D-5E        ; omit _
+                    / %x60           ; omit lowercase letters and { | }
+                    / %x7E-D7FF      ; omit surrogates
+                    / %xE000-10FFFF
+reserved-char       = non-name-char / ALPHA / DIGIT / "-" / "." / ":" / "_"
```
And updating the fallback behavior to again emit `{�}` when the first `1*name-char` doesn't match.

Reply:
Would it be possible to leave the `reserved-*` syntax updates out of this PR, and handle them separately?

Reply:
Done.